artifact.go

```diff
@@ -267,24 +268,37 @@ func (m artifactModule) postTestResultDataHandler(c *gin.Context) {
 	for _, artifact := range artifacts {
 		summary, details, err := getTestSummaryAndDetails(artifact.Data, artifact.ArtifactID, buildID)
 		if err != nil {
-			log.Warn().
+			logEvent := log.Warn().
 				WithError(err).
 				WithString("filename", artifact.Name).
 				WithUint("build", buildID).
-				WithUint("artifact", artifact.ArtifactID).
-				Message("Failed to unmarshal; invalid TRX/XML format, or the structure" +
-					" receiving the data is malformed.")
-
-			ginutil.WriteProblemError(c, err,
-				problem.Response{
-					Type:   "/prob/unexpected-response-format",
-					Status: 502,
-					Title:  "Unexpected response format.",
-					Detail: fmt.Sprintf(
-						"Failed parsing test result ID %d, for build with ID %d in"+
-							" database. Expected TRX/XML format. Could also be due to an"+
-							" internal bug.", summary.ArtifactID, buildID),
-				})
+				WithUint("artifact", artifact.ArtifactID)
+			if syntaxErr := err.(*xml.SyntaxError); syntaxErr != nil || strings.HasPrefix(err.Error(), "xml:") {
```

Comment:
This cast will fail if the `syntaxErr` is not of the type `*xml.SyntaxError`.

A safer bet when casting is to use the second return value from the cast, like so:

```go
var err error = errors.New("foo")

syntaxErr := err.(*xml.SyntaxError) // panic

syntaxErr, ok := err.(*xml.SyntaxError) // ok = false
```

But [since Go v1.13 you should instead use `errors.As` and `errors.Is` for these cases](https://go.dev/blog/go1.13-errors), as they handle nested errors

```suggestion
			var syntaxErr *xml.SyntaxError
			if errors.As(err, &syntaxErr) {
```